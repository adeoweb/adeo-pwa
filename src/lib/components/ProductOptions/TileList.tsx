import React, { FunctionComponent, useMemo } from 'react';
import Tile from 'src/lib/components/ProductOptions/Tile';
import { TValueListProps } from 'src/lib/components/ProductOptions/ProductOptionsTypes';

const TileList: FunctionComponent<TValueListProps> = ({
    getItemKey,
    selectedValue,
    items,
    onSelectionChange
}) => {
    const tiles = useMemo(
        () =>
            items.map(item => {
                const isSelected =
                    Boolean(selectedValue.label) &&
                    item.label === selectedValue.label;

                return (
                    <Tile
                        key={getItemKey(item)}
                        isSelected={isSelected}
                        item={item}
                        onClick={onSelectionChange}
                    />
                );
            }),
        [getItemKey, selectedValue.label, items, onSelectionChange]
    );

    return <ul className="config-size-list">{tiles}</ul>;
};

TileList.displayName = 'TileList';

export default TileList;
