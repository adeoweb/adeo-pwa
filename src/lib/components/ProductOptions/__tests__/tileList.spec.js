import React from 'react';
import testRenderer from 'react-test-renderer';

import TileList from '../TileList';

const defaultProps = {
    items: [
        {
            id: '1',
            label: 'foo'
        }
    ],
    selectedValue: {},
    getItemKey: jest.fn(),
    onSelectionChange: jest.fn()
};

test('renders TileList component correctly', () => {
    const component = testRenderer.create(<TileList {...defaultProps} />);
    expect(component.toJSON()).toMatchSnapshot();
});
