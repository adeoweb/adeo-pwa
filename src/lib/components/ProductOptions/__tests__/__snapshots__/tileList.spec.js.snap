// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TileList component correctly 1`] = `
<ul
  className="config-size-list"
>
  <li
    className=""
  >
    <button
      onClick={[Function]}
    >
      foo
    </button>
  </li>
</ul>
`;
